### ISSUE TITLE
Numeric Mask right to left

### ISSUE THREAD

**Issue by joaolongo** [RedMadRobot/input-mask-ios#25]
Hello, I need to implement a simple decimal mask with the following format:

`999.999.999`

When I type my text right now, I get the following result:

`123.456.78 (left to right)`

I need to type my numbers and the mask to work from right to left, so when I type those numbers I get the following result:

`12.345.678`

Is this possible?

**Comment by DamascenoRafael** [comment]
@joaolongo same here. Did you find any solution?

Using format: `[90000]-[0000]`
Getting: `12345-678`
expected: `1234-5678`

**Comment by taflanidi** [comment]
Hi @joaolongo and @DamascenoRafael,
Thanks for your question.

I think you should consider subclassing the `MaskedTextFieldDelegateListener` for your needs, overriding these methods:
```swift
open func deleteText(
    inRange range: NSRange,
    inField field: UITextField
) -> (String, Bool)
```
and 
```swift
open func modifyText(
    inRange range: NSRange,
    inField field: UITextField,
    withText text: String
) -> (String, Bool)
```

You need to reverse the [original text](https://github.com/RedMadRobot/input-mask-ios/blob/master/Source/InputMask/InputMask/Classes/View/MaskedTextFieldDelegate.swift#L227) before applying the mask, and then reverse [the result](https://github.com/RedMadRobot/input-mask-ios/blob/master/Source/InputMask/InputMask/Classes/View/MaskedTextFieldDelegate.swift#L233) before assignment to the text field.

I didn't have an opportunity to try this myself, unfortunately. So, should you have any clarifying questions, please go ahead and ask them.

**Comment by joaolongo** [comment]
@DamascenoRafael 
I did find a solution to this issue, but I had to implement a new pattern myself.
Here, try this:

[https://gist.github.com/joaolongo/957a6e2826ce6edc0ad531d2f1081db3](url)

**Comment by lutchobandeira** [comment]
I have the same use case. It would be nice to have this feature implemented in the lib.

**Comment by taflanidi** [comment]
Okay, I'm considering this as a feature candidate for future releases.
Please, stay tuned.

**Comment by taflanidi** [comment]
Implemented in `3.4.0`.

Use `ReversedMaskTextFieldDelegate` for your needs. 

Unfortunately, you won't be able to use decimal format yet. I'll need some time to investigate that.
